docs/standard/serialization/system-text-json/snippets/how-to/csharp/Utf8ReaderPartialRead.cs

```diff
@@ -7,74 +7,85 @@ public class Utf8ReaderPartialRead
     {
         public static void Run()
         {
-            var jsonString = @"{
-                ""Date"": ""2019-08-01T00:00:00-07:00"",
-                ""Temperature"": 25,
-                ""TemperatureRanges"": {
-                    ""Cold"": { ""High"": 20, ""Low"": -10 },
-                    ""Hot"": { ""High"": 60, ""Low"": 20 }
-                },
-                ""Summary"": ""Hot"",
-            }";
 
-            byte[] bytes = Encoding.UTF8.GetBytes(jsonString);
-            var stream = new MemoryStream(bytes);
-
-            var buffer = new byte[4096];
+            var jsonString = """
+    {
+        "Date": "2019-08-01T00:00:00-07:00",
+        "Temperature": 25,
+        "TemperatureRanges": {
+            "Cold": { "High": 20, "Low": -10 },
+            "Hot": { "High": 60, "Low": 20
+        },
+        "Summary": "Hot"
+    }
+    """u8;
 
-            // Fill the buffer.
-            // For this snippet, we're assuming the stream is open and has data.
-            // If it might be closed or empty, check if the return value is 0.
-            stream.Read(buffer);
+            // Buffer size wouldn't typically be this small
+            // Using a small size to demonstrate refilling buffer given small payload
+            const int SIZE = 64;
+            var searchText = "Summary"u8;
+            var foundTerm = false;
+            var stream = new MemoryStream(jsonString.ToArray());
+            Span<byte> buffer = new byte[SIZE];
+            int read = stream.Read(buffer);
 
-            // We set isFinalBlock to false since we expect more data in a subsequent read from the stream.
-            var reader = new Utf8JsonReader(buffer, isFinalBlock: false, state: default);
-            Console.WriteLine($"String in buffer is: {Encoding.UTF8.GetString(buffer)}");
+            var reader = new Utf8JsonReader(buffer[0..read], isFinalBlock: read < SIZE, state: default);
```

Comment:
I'm guessing that the buffer size has been calibrated such that `isFinalBlock` is always false? Can we clarify this in a comment?

Reply:
Ping @richlander